tools/cbmc/proofs/DNS/DNSgetHostByName_a/DNSgetHostByName_a_harness.c

```diff
@@ -0,0 +1,62 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_DNS.h"
+#include "FreeRTOS_IP_Private.h"
+
+
+/*This assumes that prvParseDNSReply, FreeRTOS_socket, FreeRTOS_GetUDPPayloadBuffer, pvPortMalloc and xTaskResumeAll
+are implemented correctly. The function func mimics the callback function.*/
+
+uint32_t prvParseDNSReply( uint8_t *pucUDPPayloadBuffer, size_t xBufferLength, TickType_t xIdentifier) { }
+
+Socket_t FreeRTOS_socket( BaseType_t xDomain, BaseType_t xType, BaseType_t xProtocol ){
+	Socket_t xCreatedSocket = malloc(sizeof(Socket_t));
+	return xCreatedSocket;
+}
+
+void * FreeRTOS_GetUDPPayloadBuffer(size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks ) {
+	void *pvReturn = malloc(50);
+	return pvReturn;
+}
+
+void *pvPortMalloc(size_t xWantedSize) {
+	void *pvSize = malloc(xWantedSize);
+	return pvSize;
```

Comment:
I'm not requesting any changes for this now, but a broader discussion we need to have is: can malloc fail. If it can, we should be ensuring memory safety even in that case.

We might build up a nice library of utility functions, similar to what aws-c-common has, that includes a might-fail mallocator.